packages/flutter/lib/src/material/data_table.dart

```diff
@@ -383,7 +384,12 @@ class DataTable extends StatelessWidget {
     this.onSelectAll,
     this.decoration,
     this.dataRowColor,
+    @Deprecated(
+      'Use dataRowHeightSettings instead. '
+      'This feature was deprecated after v2.13.0-0.4.pre.',
+    )
     this.dataRowHeight,
+    this.dataRowHeightSettings,
```

Comment:
`Settings` seems like a weird name and is not use elsewhere in flutter.

Some ideas: `dataRowHeightStyle`, `dataRowHeightMode`

Reply:
also this constructor can initialize dataRowHeightSettings from dataRowHeight if dataRowHeightSettings is not provider
```dart
DataTable({
  ...
  DataRowHeightSettings dataRowHeightSettings
}) : this.dataRowHeightSettings = dataRowHeightSettings ?? DataRowHeightSettings.fixed(height: dataRowHeight);
```

So the rest of the class can just deal with dataRowHeightSettings